{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "wicked-malawi",
   "metadata": {},
   "source": [
    "## Install Swan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "periodic-devon",
   "metadata": {},
   "outputs": [],
   "source": [
    "!git clone git@github.com:mortazavilab/swan_vis.git\n",
    "!cd swan_vis\n",
    "!pip install ."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "received-merchandise",
   "metadata": {},
   "source": [
    "Alternatively...."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dried-fortune",
   "metadata": {},
   "outputs": [],
   "source": [
    "!pip install swan_vis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "concerned-connecticut",
   "metadata": {},
   "source": [
    "## Already run \n",
    "The following section was already run so you don't have to wait for results, but you're welcome to give it a try!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "intimate-certificate",
   "metadata": {},
   "outputs": [],
   "source": [
    "# data download\n",
    "!wget http://crick.bio.uci.edu/freese/210830_ucd_worksho/data.tgz\n",
    "!tar -xf data.tgz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "plain-superintendent",
   "metadata": {},
   "outputs": [],
   "source": [
    "import swan_vis as swan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "invisible-republican",
   "metadata": {},
   "outputs": [],
   "source": [
    "# construct SwanGraph\n",
    "annot_gtf = 'ref/gencode.vM21.annotation.gtf'\n",
    "data_gtf = 'data/all_talon_observedOnly.gtf'\n",
    "ab_file = 'data/all_talon_abundance_filtered.tsv'\n",
    "meta = 'data/metadata.tsv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "valued-constant",
   "metadata": {},
   "outputs": [],
   "source": [
    "sg = swan.SwanGraph()\n",
    "sg.add_annotation(annot_gtf)\n",
    "sg.add_transcriptome(data_gtf)\n",
    "sg.add_abundance(ab_file)\n",
    "sg.add_metadata(meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "superb-designation",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saving graph as swan.p\n"
     ]
    }
   ],
   "source": [
    "sg.save_graph('swan')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "legal-european",
   "metadata": {},
   "outputs": [],
   "source": [
    "# compare 0 and 72 hr time point \n",
    "obs_col = 'time_point'\n",
    "obs_conditions = ['0hr', '72hr']\n",
    "_ = sg.de_gene_test(obs_col, obs_conditions=obs_conditions)\n",
    "_ = sg.de_transcript_test(obs_col, obs_conditions=obs_conditions)\n",
    "_ = sg.die_gene_test(obs_col, obs_conditions=obs_conditions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "internal-equilibrium",
   "metadata": {},
   "outputs": [],
   "source": [
    "# find novel exon-skipping and intron retention events\n",
    "_ = sg.find_es_genes()\n",
    "_ = sg.find_ir_genes()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "legislative-frank",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save graph again so we have access to all the results \n",
    "sg.save_graph('swan')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "special-giant",
   "metadata": {},
   "source": [
    "## Your turn!"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "numerical-african",
   "metadata": {},
   "source": [
    "Now we'll take a look at some of the results from Swan and make some cool plots. If you want to do anything off-book or are confused about anything please take a look at the [documentation](https://freese.gitbook.io/swan/). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "noticed-healthcare",
   "metadata": {},
   "outputs": [],
   "source": [
    "# download the SwanGraph\n",
    "!wget http://crick.bio.uci.edu/freese/210830_ucd_workshop/swan.p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "little-ethnic",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read in graph from swan.p\n"
     ]
    }
   ],
   "source": [
    "# load the SwanGraph\n",
    "sg = swan.read('swan.p')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bigger-finance",
   "metadata": {},
   "outputs": [],
   "source": [
    "# TODO - start with IR / ES stuff that visualizing the smol graphs is useful for"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "accepted-detector",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add some colors to represent the 2 different time points\n",
    "green = '#019f73'\n",
    "pink = '#cb79a7'\n",
    "cmap = {'0hr': pink, '72hr': green}\n",
    "sg.set_metadata_colors('time_point', cmap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "medieval-enterprise",
   "metadata": {},
   "outputs": [],
   "source": [
    "# results from differential gene expression test\n",
    "obs_col = 'time_point'\n",
    "obs_conditions = ['0hr', '72hr']\n",
    "de_genes = sg.get_de_genes(obs_col, obs_conditions=obs_conditions,\n",
    "                           q=0.05, log2fc=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "disturbed-truck",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gid</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "      <th>log2fc</th>\n",
       "      <th>mean</th>\n",
       "      <th>zero_mean</th>\n",
       "      <th>grad</th>\n",
       "      <th>coef_mle</th>\n",
       "      <th>coef_sd</th>\n",
       "      <th>ll</th>\n",
       "      <th>gname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19866</th>\n",
       "      <td>ENSMUSG00000070733.13</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>0.561909</td>\n",
       "      <td>False</td>\n",
       "      <td>0.500002</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-16.217098</td>\n",
       "      <td>Fryl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18848</th>\n",
       "      <td>ENSMUSG00000065852.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>0.407021</td>\n",
       "      <td>False</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-15.435992</td>\n",
       "      <td>Gm22353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18852</th>\n",
       "      <td>ENSMUSG00000065862.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>0.407021</td>\n",
       "      <td>False</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-15.435992</td>\n",
       "      <td>Gm24029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19949</th>\n",
       "      <td>ENSMUSG00000071042.12</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>0.203510</td>\n",
       "      <td>False</td>\n",
       "      <td>0.409290</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-11.215066</td>\n",
       "      <td>Rasgrp3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7466</th>\n",
       "      <td>ENSMUSG00000029699.13</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>0.578117</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-28.840319</td>\n",
       "      <td>Ssc4d</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         gid  pval  qval      log2fc      mean  zero_mean  \\\n",
       "19866  ENSMUSG00000070733.13   0.0   0.0  283.913085  0.561909      False   \n",
       "18848   ENSMUSG00000065852.1   0.0   0.0  283.913085  0.407021      False   \n",
       "18852   ENSMUSG00000065862.1   0.0   0.0  283.913085  0.407021      False   \n",
       "19949  ENSMUSG00000071042.12   0.0   0.0  283.913085  0.203510      False   \n",
       "7466   ENSMUSG00000029699.13   0.0   0.0  283.913085  0.578117      False   \n",
       "\n",
       "           grad    coef_mle        coef_sd         ll    gname  \n",
       "19866  0.500002  283.913085  2.222759e-162 -16.217098     Fryl  \n",
       "18848  0.500000  283.913085  2.222759e-162 -15.435992  Gm22353  \n",
       "18852  0.500000  283.913085  2.222759e-162 -15.435992  Gm24029  \n",
       "19949  0.409290  283.913085  2.222759e-162 -11.215066  Rasgrp3  \n",
       "7466   1.000000  283.913085  2.222759e-162 -28.840319    Ssc4d  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "de_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "geological-walter",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gid</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "      <th>log2fc</th>\n",
       "      <th>mean</th>\n",
       "      <th>zero_mean</th>\n",
       "      <th>grad</th>\n",
       "      <th>coef_mle</th>\n",
       "      <th>coef_sd</th>\n",
       "      <th>ll</th>\n",
       "      <th>gname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19415</th>\n",
       "      <td>ENSMUSG00000068697.7</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>4.689763</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-38.567663</td>\n",
       "      <td>Myoz1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6503</th>\n",
       "      <td>ENSMUSG00000028116.13</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>28.415941</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-45.865588</td>\n",
       "      <td>Myoz2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20253</th>\n",
       "      <td>ENSMUSG00000072720.9</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>5.471390</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-39.200583</td>\n",
       "      <td>Myo18b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10601</th>\n",
       "      <td>ENSMUSG00000037139.15</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>16.414171</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-43.658559</td>\n",
       "      <td>Myom3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8369</th>\n",
       "      <td>ENSMUSG00000031461.4</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>351.204727</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-55.932105</td>\n",
       "      <td>Myom2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4263</th>\n",
       "      <td>ENSMUSG00000024049.15</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>20.664499</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-44.587521</td>\n",
       "      <td>Myom1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4505</th>\n",
       "      <td>ENSMUSG00000024471.12</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>10.071095</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-41.596729</td>\n",
       "      <td>Myot</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5560</th>\n",
       "      <td>ENSMUSG00000026459.5</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2162.365112</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-63.212547</td>\n",
       "      <td>Myog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3582</th>\n",
       "      <td>ENSMUSG00000022272.17</td>\n",
       "      <td>0.00066</td>\n",
       "      <td>0.004051</td>\n",
       "      <td>1.475328</td>\n",
       "      <td>8.793978</td>\n",
       "      <td>False</td>\n",
       "      <td>0.689040</td>\n",
       "      <td>1.475328</td>\n",
       "      <td>4.331994e-01</td>\n",
       "      <td>-3.089287</td>\n",
       "      <td>Myo10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         gid     pval      qval      log2fc         mean  \\\n",
       "19415   ENSMUSG00000068697.7  0.00000  0.000000  283.913085     4.689763   \n",
       "6503   ENSMUSG00000028116.13  0.00000  0.000000  283.913085    28.415941   \n",
       "20253   ENSMUSG00000072720.9  0.00000  0.000000  283.913085     5.471390   \n",
       "10601  ENSMUSG00000037139.15  0.00000  0.000000  283.913085    16.414171   \n",
       "8369    ENSMUSG00000031461.4  0.00000  0.000000  283.913085   351.204727   \n",
       "4263   ENSMUSG00000024049.15  0.00000  0.000000  283.913085    20.664499   \n",
       "4505   ENSMUSG00000024471.12  0.00000  0.000000  283.913085    10.071095   \n",
       "5560    ENSMUSG00000026459.5  0.00000  0.000000  283.913085  2162.365112   \n",
       "3582   ENSMUSG00000022272.17  0.00066  0.004051    1.475328     8.793978   \n",
       "\n",
       "       zero_mean      grad    coef_mle        coef_sd         ll   gname  \n",
       "19415      False  1.000000  283.913085  2.222759e-162 -38.567663   Myoz1  \n",
       "6503       False  0.999999  283.913085  2.222759e-162 -45.865588   Myoz2  \n",
       "20253      False  1.000000  283.913085  2.222759e-162 -39.200583  Myo18b  \n",
       "10601      False  0.999999  283.913085  2.222759e-162 -43.658559   Myom3  \n",
       "8369       False  0.999999  283.913085  2.222759e-162 -55.932105   Myom2  \n",
       "4263       False  0.999999  283.913085  2.222759e-162 -44.587521   Myom1  \n",
       "4505       False  0.999999  283.913085  2.222759e-162 -41.596729    Myot  \n",
       "5560       False  0.999999  283.913085  2.222759e-162 -63.212547    Myog  \n",
       "3582       False  0.689040    1.475328   4.331994e-01  -3.089287   Myo10  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "de_genes.loc[de_genes.gname.str.contains('Myo')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "indie-memorabilia",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5392"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(de_genes.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "positive-michael",
   "metadata": {},
   "outputs": [],
   "source": [
    "# results from differential transcript expression test\n",
    "de_transcripts = sg.get_de_transcripts(obs_col, obs_conditions=obs_conditions,\n",
    "                           q=0.05, log2fc=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "arbitrary-candle",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tid</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "      <th>log2fc</th>\n",
       "      <th>mean</th>\n",
       "      <th>zero_mean</th>\n",
       "      <th>grad</th>\n",
       "      <th>coef_mle</th>\n",
       "      <th>coef_sd</th>\n",
       "      <th>ll</th>\n",
       "      <th>gid</th>\n",
       "      <th>gname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>90372</th>\n",
       "      <td>ENSMUST00000184224.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>0.187303</td>\n",
       "      <td>False</td>\n",
       "      <td>0.374605</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-10.301695</td>\n",
       "      <td>ENSMUSG00000098716.1</td>\n",
       "      <td>Gm28033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90689</th>\n",
       "      <td>ENSMUST00000184591.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.312467</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-35.526327</td>\n",
       "      <td>ENSMUSG00000033182.12</td>\n",
       "      <td>Kbtbd12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90287</th>\n",
       "      <td>ENSMUST00000184135.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>0.203510</td>\n",
       "      <td>False</td>\n",
       "      <td>0.409290</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-11.215066</td>\n",
       "      <td>ENSMUSG00000098761.1</td>\n",
       "      <td>Gm18821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16680</th>\n",
       "      <td>ENSMUST00000079773.13</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>13.458757</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-42.859372</td>\n",
       "      <td>ENSMUSG00000003345.16</td>\n",
       "      <td>Csnk1g2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45913</th>\n",
       "      <td>ENSMUST00000132080.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>0.187303</td>\n",
       "      <td>False</td>\n",
       "      <td>0.374605</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-10.301695</td>\n",
       "      <td>ENSMUSG00000013858.14</td>\n",
       "      <td>Tmem259</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         tid  pval  qval      log2fc       mean  zero_mean  \\\n",
       "90372   ENSMUST00000184224.1   0.0   0.0  283.913085   0.187303      False   \n",
       "90689   ENSMUST00000184591.1   0.0   0.0  283.913085   2.312467      False   \n",
       "90287   ENSMUST00000184135.1   0.0   0.0  283.913085   0.203510      False   \n",
       "16680  ENSMUST00000079773.13   0.0   0.0  283.913085  13.458757      False   \n",
       "45913   ENSMUST00000132080.1   0.0   0.0  283.913085   0.187303      False   \n",
       "\n",
       "           grad    coef_mle        coef_sd         ll                    gid  \\\n",
       "90372  0.374605  283.913085  2.222759e-162 -10.301695   ENSMUSG00000098716.1   \n",
       "90689  0.999999  283.913085  2.222759e-162 -35.526327  ENSMUSG00000033182.12   \n",
       "90287  0.409290  283.913085  2.222759e-162 -11.215066   ENSMUSG00000098761.1   \n",
       "16680  0.999999  283.913085  2.222759e-162 -42.859372  ENSMUSG00000003345.16   \n",
       "45913  0.374605  283.913085  2.222759e-162 -10.301695  ENSMUSG00000013858.14   \n",
       "\n",
       "         gname  \n",
       "90372  Gm28033  \n",
       "90689  Kbtbd12  \n",
       "90287  Gm18821  \n",
       "16680  Csnk1g2  \n",
       "45913  Tmem259  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "de_transcripts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "neither-advocacy",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tid</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "      <th>log2fc</th>\n",
       "      <th>mean</th>\n",
       "      <th>zero_mean</th>\n",
       "      <th>grad</th>\n",
       "      <th>coef_mle</th>\n",
       "      <th>coef_sd</th>\n",
       "      <th>ll</th>\n",
       "      <th>gid</th>\n",
       "      <th>gname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>89607</th>\n",
       "      <td>ENSMUST00000183377.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>0.203510</td>\n",
       "      <td>False</td>\n",
       "      <td>0.409290</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-11.215066</td>\n",
       "      <td>ENSMUSG00000033577.18</td>\n",
       "      <td>Myo6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49005</th>\n",
       "      <td>ENSMUST00000135375.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>1.563254</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000001</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-33.923097</td>\n",
       "      <td>ENSMUSG00000000631.20</td>\n",
       "      <td>Myo18a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86409</th>\n",
       "      <td>ENSMUST00000179759.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>20.477194</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-44.551066</td>\n",
       "      <td>ENSMUSG00000024049.15</td>\n",
       "      <td>Myom1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45155</th>\n",
       "      <td>ENSMUST00000131251.1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>0.390814</td>\n",
       "      <td>False</td>\n",
       "      <td>0.781626</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-21.519703</td>\n",
       "      <td>ENSMUSG00000031461.4</td>\n",
       "      <td>Myom2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58333</th>\n",
       "      <td>ENSMUST00000145304.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>10.364664</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-41.799408</td>\n",
       "      <td>ENSMUSG00000037139.15</td>\n",
       "      <td>Myom3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        tid  pval  qval      log2fc       mean  zero_mean  \\\n",
       "89607  ENSMUST00000183377.1   0.0   0.0  283.913085   0.203510      False   \n",
       "49005  ENSMUST00000135375.3   0.0   0.0  283.913085   1.563254      False   \n",
       "86409  ENSMUST00000179759.2   0.0   0.0  283.913085  20.477194      False   \n",
       "45155  ENSMUST00000131251.1   0.0   0.0  283.913085   0.390814      False   \n",
       "58333  ENSMUST00000145304.7   0.0   0.0  283.913085  10.364664      False   \n",
       "\n",
       "           grad    coef_mle        coef_sd         ll                    gid  \\\n",
       "89607  0.409290  283.913085  2.222759e-162 -11.215066  ENSMUSG00000033577.18   \n",
       "49005  1.000001  283.913085  2.222759e-162 -33.923097  ENSMUSG00000000631.20   \n",
       "86409  0.999999  283.913085  2.222759e-162 -44.551066  ENSMUSG00000024049.15   \n",
       "45155  0.781626  283.913085  2.222759e-162 -21.519703   ENSMUSG00000031461.4   \n",
       "58333  1.000000  283.913085  2.222759e-162 -41.799408  ENSMUSG00000037139.15   \n",
       "\n",
       "        gname  \n",
       "89607    Myo6  \n",
       "49005  Myo18a  \n",
       "86409   Myom1  \n",
       "45155   Myom2  \n",
       "58333   Myom3  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "de_transcripts.loc[de_transcripts.gname.str.contains('Myo')].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "human-status",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tid</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "      <th>log2fc</th>\n",
       "      <th>mean</th>\n",
       "      <th>zero_mean</th>\n",
       "      <th>grad</th>\n",
       "      <th>coef_mle</th>\n",
       "      <th>coef_sd</th>\n",
       "      <th>ll</th>\n",
       "      <th>gid</th>\n",
       "      <th>gname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>146970</th>\n",
       "      <td>TALONT000524508</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>4.502460</td>\n",
       "      <td>False</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-38.394349</td>\n",
       "      <td>ENSMUSG00000032294.17</td>\n",
       "      <td>Pkm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146957</th>\n",
       "      <td>TALONT000523760</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>2.987293</td>\n",
       "      <td>13.641756</td>\n",
       "      <td>False</td>\n",
       "      <td>0.352200</td>\n",
       "      <td>2.987293</td>\n",
       "      <td>6.332595e-01</td>\n",
       "      <td>-4.011946</td>\n",
       "      <td>ENSMUSG00000032294.17</td>\n",
       "      <td>Pkm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146953</th>\n",
       "      <td>TALONT000523686</td>\n",
       "      <td>0.006228</td>\n",
       "      <td>0.042511</td>\n",
       "      <td>2.183307</td>\n",
       "      <td>4.312392</td>\n",
       "      <td>False</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>2.183307</td>\n",
       "      <td>7.981339e-01</td>\n",
       "      <td>-6.115639</td>\n",
       "      <td>ENSMUSG00000032294.17</td>\n",
       "      <td>Pkm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146951</th>\n",
       "      <td>TALONT000523682</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>1.360462</td>\n",
       "      <td>14.967567</td>\n",
       "      <td>False</td>\n",
       "      <td>1.263405</td>\n",
       "      <td>1.360462</td>\n",
       "      <td>3.206211e-01</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSMUSG00000032294.17</td>\n",
       "      <td>Pkm</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    tid      pval      qval      log2fc       mean  zero_mean  \\\n",
       "146970  TALONT000524508  0.000000  0.000000  283.913085   4.502460      False   \n",
       "146957  TALONT000523760  0.000002  0.000020    2.987293  13.641756      False   \n",
       "146953  TALONT000523686  0.006228  0.042511    2.183307   4.312392      False   \n",
       "146951  TALONT000523682  0.000022  0.000176    1.360462  14.967567      False   \n",
       "\n",
       "            grad    coef_mle        coef_sd         ll                    gid  \\\n",
       "146970  0.999999  283.913085  2.222759e-162 -38.394349  ENSMUSG00000032294.17   \n",
       "146957  0.352200    2.987293   6.332595e-01  -4.011946  ENSMUSG00000032294.17   \n",
       "146953  0.000038    2.183307   7.981339e-01  -6.115639  ENSMUSG00000032294.17   \n",
       "146951  1.263405    1.360462   3.206211e-01   0.000000  ENSMUSG00000032294.17   \n",
       "\n",
       "       gname  \n",
       "146970   Pkm  \n",
       "146957   Pkm  \n",
       "146953   Pkm  \n",
       "146951   Pkm  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "de_transcripts.loc[de_transcripts.gname == 'Pkm']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "cooked-morning",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tid</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "      <th>log2fc</th>\n",
       "      <th>mean</th>\n",
       "      <th>zero_mean</th>\n",
       "      <th>grad</th>\n",
       "      <th>coef_mle</th>\n",
       "      <th>coef_sd</th>\n",
       "      <th>ll</th>\n",
       "      <th>gid</th>\n",
       "      <th>gname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>145514</th>\n",
       "      <td>TALONT000415087</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>10.503345</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-41.851966</td>\n",
       "      <td>ENSMUSG00000028464.16</td>\n",
       "      <td>Tpm2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145516</th>\n",
       "      <td>TALONT000415089</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>6.155773</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-39.562596</td>\n",
       "      <td>ENSMUSG00000028464.16</td>\n",
       "      <td>Tpm2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145517</th>\n",
       "      <td>TALONT000415177</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.141371</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000004</td>\n",
       "      <td>283.913085</td>\n",
       "      <td>2.222759e-162</td>\n",
       "      <td>-35.287652</td>\n",
       "      <td>ENSMUSG00000028464.16</td>\n",
       "      <td>Tpm2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26461</th>\n",
       "      <td>ENSMUST00000107913.9</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.570377</td>\n",
       "      <td>673.755588</td>\n",
       "      <td>False</td>\n",
       "      <td>5.430085</td>\n",
       "      <td>3.570377</td>\n",
       "      <td>1.180513e-01</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSMUSG00000028464.16</td>\n",
       "      <td>Tpm2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145515</th>\n",
       "      <td>TALONT000415088</td>\n",
       "      <td>0.003188</td>\n",
       "      <td>0.022426</td>\n",
       "      <td>3.218529</td>\n",
       "      <td>5.673465</td>\n",
       "      <td>False</td>\n",
       "      <td>0.091297</td>\n",
       "      <td>3.218529</td>\n",
       "      <td>1.091389e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>ENSMUSG00000028464.16</td>\n",
       "      <td>Tpm2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         tid      pval      qval      log2fc        mean  \\\n",
       "145514       TALONT000415087  0.000000  0.000000  283.913085   10.503345   \n",
       "145516       TALONT000415089  0.000000  0.000000  283.913085    6.155773   \n",
       "145517       TALONT000415177  0.000000  0.000000  283.913085    2.141371   \n",
       "26461   ENSMUST00000107913.9  0.000000  0.000000    3.570377  673.755588   \n",
       "145515       TALONT000415088  0.003188  0.022426    3.218529    5.673465   \n",
       "\n",
       "        zero_mean      grad    coef_mle        coef_sd         ll  \\\n",
       "145514      False  1.000000  283.913085  2.222759e-162 -41.851966   \n",
       "145516      False  1.000000  283.913085  2.222759e-162 -39.562596   \n",
       "145517      False  1.000004  283.913085  2.222759e-162 -35.287652   \n",
       "26461       False  5.430085    3.570377   1.180513e-01   0.000000   \n",
       "145515      False  0.091297    3.218529   1.091389e+00   0.000000   \n",
       "\n",
       "                          gid gname  \n",
       "145514  ENSMUSG00000028464.16  Tpm2  \n",
       "145516  ENSMUSG00000028464.16  Tpm2  \n",
       "145517  ENSMUSG00000028464.16  Tpm2  \n",
       "26461   ENSMUSG00000028464.16  Tpm2  \n",
       "145515  ENSMUSG00000028464.16  Tpm2  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "de_transcripts.loc[de_transcripts.gname == 'Tpm2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "senior-tuesday",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12232"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(de_transcripts.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "behind-demand",
   "metadata": {},
   "outputs": [],
   "source": [
    "# results from differential transcript expression test\n",
    "die_genes = sg.get_die_genes(obs_col=obs_col, obs_conditions=obs_conditions,\n",
    "                             p=0.05, dpi=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "amateur-filename",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gid</th>\n",
       "      <th>p_val</th>\n",
       "      <th>dpi</th>\n",
       "      <th>adj_p_val</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSMUSG00000061689.15</td>\n",
       "      <td>1.443605e-19</td>\n",
       "      <td>67.565826</td>\n",
       "      <td>3.424032e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSMUSG00000020152.7</td>\n",
       "      <td>6.042619e-11</td>\n",
       "      <td>19.780212</td>\n",
       "      <td>8.726454e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENSMUSG00000000326.13</td>\n",
       "      <td>9.814141e-04</td>\n",
       "      <td>28.234030</td>\n",
       "      <td>5.577975e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENSMUSG00000000420.15</td>\n",
       "      <td>1.257904e-29</td>\n",
       "      <td>37.495850</td>\n",
       "      <td>4.475362e-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENSMUSG00000000440.12</td>\n",
       "      <td>9.107806e-03</td>\n",
       "      <td>44.185776</td>\n",
       "      <td>4.001738e-02</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      gid         p_val        dpi     adj_p_val\n",
       "3   ENSMUSG00000061689.15  1.443605e-19  67.565826  3.424032e-18\n",
       "5    ENSMUSG00000020152.7  6.042619e-11  19.780212  8.726454e-10\n",
       "11  ENSMUSG00000000326.13  9.814141e-04  28.234030  5.577975e-03\n",
       "17  ENSMUSG00000000420.15  1.257904e-29  37.495850  4.475362e-28\n",
       "18  ENSMUSG00000000440.12  9.107806e-03  44.185776  4.001738e-02"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "die_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "harmful-guatemala",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add gene names, which I should make automatic in future releases!\n",
    "gnames = sg.t_df[['gname', 'gid']].drop_duplicates()\n",
    "die_genes = die_genes.merge(gnames, on='gid', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "junior-hazard",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gid</th>\n",
       "      <th>p_val</th>\n",
       "      <th>dpi</th>\n",
       "      <th>adj_p_val</th>\n",
       "      <th>gname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>ENSMUSG00000026414.13</td>\n",
       "      <td>5.312136e-237</td>\n",
       "      <td>98.948153</td>\n",
       "      <td>2.645927e-234</td>\n",
       "      <td>Tnnt2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>ENSMUSG00000020836.15</td>\n",
       "      <td>1.787333e-64</td>\n",
       "      <td>97.932594</td>\n",
       "      <td>1.718034e-62</td>\n",
       "      <td>Coro6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>ENSMUSG00000020439.17</td>\n",
       "      <td>5.141485e-75</td>\n",
       "      <td>97.710327</td>\n",
       "      <td>5.523568e-73</td>\n",
       "      <td>Smtn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>946</th>\n",
       "      <td>ENSMUSG00000039542.16</td>\n",
       "      <td>7.076545e-128</td>\n",
       "      <td>96.704330</td>\n",
       "      <td>1.436015e-125</td>\n",
       "      <td>Ncam1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>ENSMUSG00000036918.16</td>\n",
       "      <td>3.520068e-06</td>\n",
       "      <td>95.447243</td>\n",
       "      <td>3.115743e-05</td>\n",
       "      <td>Ttc7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       gid          p_val        dpi      adj_p_val  gname\n",
       "281  ENSMUSG00000026414.13  5.312136e-237  98.948153  2.645927e-234  Tnnt2\n",
       "161  ENSMUSG00000020836.15   1.787333e-64  97.932594   1.718034e-62  Coro6\n",
       "151  ENSMUSG00000020439.17   5.141485e-75  97.710327   5.523568e-73   Smtn\n",
       "946  ENSMUSG00000039542.16  7.076545e-128  96.704330  1.436015e-125  Ncam1\n",
       "496  ENSMUSG00000036918.16   3.520068e-06  95.447243   3.115743e-05   Ttc7"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "die_genes = die_genes.sort_values(by='dpi', ascending=False)\n",
    "die_genes.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ranging-awareness",
   "metadata": {},
   "source": [
    "## Make some visualizations of the analysis results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "solid-worst",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir figures"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "patent-offer",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Plotting transcripts for ENSMUSG00000061689.15\n",
      "Saving transcript path graph for ENSMUST00000109566.8 as figures/dlgap4_ENSMUST00000109566.8_path.png\n",
      "Saving transcript path graph for ENSMUST00000099145.5 as figures/dlgap4_ENSMUST00000099145.5_path.png\n",
      "Saving transcript path graph for ENSMUST00000127944.8 as figures/dlgap4_ENSMUST00000127944.8_path.png\n",
      "Saving transcript path graph for ENSMUST00000131157.8 as figures/dlgap4_ENSMUST00000131157.8_path.png\n",
      "Generating report for ENSMUSG00000061689.15\n"
     ]
    }
   ],
   "source": [
    "sg.gen_report('Dlgap4',\n",
    "              'figures/dlgap4',\n",
    "              metadata_cols=['time_point'],\n",
    "              cmap='viridis',\n",
    "              transcript_name=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "affected-citizenship",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Plotting transcripts for ENSMUSG00000061689.15\n",
      "Saving transcript path graph for ENSMUST00000109566.8 as figures/dlgap4_browser_ENSMUST00000109566.8_path.png\n",
      "Saving transcript path graph for ENSMUST00000099145.5 as figures/dlgap4_browser_ENSMUST00000099145.5_path.png\n",
      "Saving transcript path graph for ENSMUST00000127944.8 as figures/dlgap4_browser_ENSMUST00000127944.8_path.png\n",
      "Saving transcript path graph for ENSMUST00000131157.8 as figures/dlgap4_browser_ENSMUST00000131157.8_path.png\n",
      "Generating report for ENSMUSG00000061689.15\n"
     ]
    }
   ],
   "source": [
    "sg.gen_report('Dlgap4',\n",
    "              'figures/dlgap4',\n",
    "              metadata_cols=['time_point'],\n",
    "              cmap='viridis',\n",
    "              transcript_name=True, \n",
    "              browser=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "corporate-heading",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Plotting transcripts for ENSMUSG00000061689.15\n",
      "Saving transcript path graph for ENSMUST00000109566.8 as figures/dlgap4_ENSMUST00000109566.8_path.png\n",
      "Saving transcript path graph for ENSMUST00000099145.5 as figures/dlgap4_ENSMUST00000099145.5_path.png\n",
      "Saving transcript path graph for ENSMUST00000127944.8 as figures/dlgap4_ENSMUST00000127944.8_path.png\n",
      "Saving transcript path graph for ENSMUST00000131157.8 as figures/dlgap4_ENSMUST00000131157.8_path.png\n",
      "Generating report for ENSMUSG00000061689.15\n"
     ]
    }
   ],
   "source": [
    "sg.gen_report('Dlgap4',\n",
    "              'figures/dlgap4',\n",
    "              metadata_cols=['time_point'],\n",
    "              cmap='magma',\n",
    "              transcript_name=True, \n",
    "              layer='pi',\n",
    "              display_numbers=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "injured-strand",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Plotting transcripts for ENSMUSG00000026414.13\n",
      "Saving transcript path graph for TALONT000394811 as figures/tnnt2_novel_TALONT000394811_path.png\n",
      "Saving transcript path graph for TALONT000394812 as figures/tnnt2_novel_TALONT000394812_path.png\n",
      "Saving transcript path graph for TALONT000396016 as figures/tnnt2_novel_TALONT000396016_path.png\n",
      "Saving transcript path graph for TALONT000396098 as figures/tnnt2_novel_TALONT000396098_path.png\n",
      "Saving transcript path graph for TALONT000394818 as figures/tnnt2_novel_TALONT000394818_path.png\n",
      "Saving transcript path graph for TALONT000394815 as figures/tnnt2_novel_TALONT000394815_path.png\n",
      "Saving transcript path graph for TALONT000394848 as figures/tnnt2_novel_TALONT000394848_path.png\n",
      "Saving transcript path graph for TALONT000394894 as figures/tnnt2_novel_TALONT000394894_path.png\n",
      "Saving transcript path graph for ENSMUST00000191055.1 as figures/tnnt2_novel_ENSMUST00000191055.1_path.png\n",
      "Saving transcript path graph for TALONT000394893 as figures/tnnt2_novel_TALONT000394893_path.png\n",
      "Saving transcript path graph for TALONT000394956 as figures/tnnt2_novel_TALONT000394956_path.png\n",
      "Saving transcript path graph for TALONT000394970 as figures/tnnt2_novel_TALONT000394970_path.png\n",
      "Saving transcript path graph for TALONT000394838 as figures/tnnt2_novel_TALONT000394838_path.png\n",
      "Generating report for ENSMUSG00000026414.13\n"
     ]
    }
   ],
   "source": [
    "sg.gen_report('Tnnt2',\n",
    "              'figures/tnnt2',\n",
    "              metadata_cols=['time_point'],\n",
    "              cmap='viridis',\n",
    "              transcript_name=True,\n",
    "              novelty=True, \n",
    "              indicate_novel=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "pacific-islam",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Plotting transcripts for ENSMUSG00000026414.13\n",
      "Saving transcript path graph for TALONT000394811 as figures/tnnt2_browser_TALONT000394811_path.png\n",
      "Saving transcript path graph for TALONT000394812 as figures/tnnt2_browser_TALONT000394812_path.png\n",
      "Saving transcript path graph for TALONT000396016 as figures/tnnt2_browser_TALONT000396016_path.png\n",
      "Saving transcript path graph for TALONT000396098 as figures/tnnt2_browser_TALONT000396098_path.png\n",
      "Saving transcript path graph for TALONT000394818 as figures/tnnt2_browser_TALONT000394818_path.png\n",
      "Saving transcript path graph for TALONT000394815 as figures/tnnt2_browser_TALONT000394815_path.png\n",
      "Saving transcript path graph for TALONT000394848 as figures/tnnt2_browser_TALONT000394848_path.png\n",
      "Saving transcript path graph for TALONT000394894 as figures/tnnt2_browser_TALONT000394894_path.png\n",
      "Saving transcript path graph for ENSMUST00000191055.1 as figures/tnnt2_browser_ENSMUST00000191055.1_path.png\n",
      "Saving transcript path graph for TALONT000394893 as figures/tnnt2_browser_TALONT000394893_path.png\n",
      "Saving transcript path graph for TALONT000394956 as figures/tnnt2_browser_TALONT000394956_path.png\n",
      "Saving transcript path graph for TALONT000394970 as figures/tnnt2_browser_TALONT000394970_path.png\n",
      "Saving transcript path graph for TALONT000394838 as figures/tnnt2_browser_TALONT000394838_path.png\n",
      "Generating report for ENSMUSG00000026414.13\n"
     ]
    }
   ],
   "source": [
    "sg.gen_report('Tnnt2',\n",
    "              'figures/tnnt2',\n",
    "              metadata_cols=['time_point'],\n",
    "              cmap='magma',\n",
    "              transcript_name=True,\n",
    "              layer='pi',\n",
    "              display_numbers=True,\n",
    "              novelty=True, \n",
    "              browser=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "prime-adrian",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Plotting transcripts for ENSMUSG00000020836.15\n",
      "Saving transcript path graph for TALONT000544061 as figures/coro6_novel_TALONT000544061_path.png\n",
      "Saving transcript path graph for ENSMUST00000102493.7 as figures/coro6_novel_ENSMUST00000102493.7_path.png\n",
      "Saving transcript path graph for TALONT000544046 as figures/coro6_novel_TALONT000544046_path.png\n",
      "Saving transcript path graph for TALONT000544055 as figures/coro6_novel_TALONT000544055_path.png\n",
      "Saving transcript path graph for TALONT000544048 as figures/coro6_novel_TALONT000544048_path.png\n",
      "Saving transcript path graph for TALONT000544058 as figures/coro6_novel_TALONT000544058_path.png\n",
      "Saving transcript path graph for TALONT000544052 as figures/coro6_novel_TALONT000544052_path.png\n",
      "Saving transcript path graph for ENSMUST00000108391.8 as figures/coro6_novel_ENSMUST00000108391.8_path.png\n",
      "Generating report for ENSMUSG00000020836.15\n"
     ]
    }
   ],
   "source": [
    "sg.gen_report('Coro6',\n",
    "              'figures/coro6',\n",
    "              metadata_cols=['time_point'],\n",
    "              cmap='viridis',\n",
    "              transcript_name=True,\n",
    "              novelty=True, \n",
    "              indicate_novel=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "blank-tanzania",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Plotting transcripts for ENSMUSG00000020836.15\n",
      "Saving transcript path graph for TALONT000544061 as figures/coro6_browser_TALONT000544061_path.png\n",
      "Saving transcript path graph for ENSMUST00000102493.7 as figures/coro6_browser_ENSMUST00000102493.7_path.png\n",
      "Saving transcript path graph for TALONT000544046 as figures/coro6_browser_TALONT000544046_path.png\n",
      "Saving transcript path graph for TALONT000544055 as figures/coro6_browser_TALONT000544055_path.png\n",
      "Saving transcript path graph for TALONT000544048 as figures/coro6_browser_TALONT000544048_path.png\n",
      "Saving transcript path graph for TALONT000544058 as figures/coro6_browser_TALONT000544058_path.png\n",
      "Saving transcript path graph for TALONT000544052 as figures/coro6_browser_TALONT000544052_path.png\n",
      "Saving transcript path graph for ENSMUST00000108391.8 as figures/coro6_browser_ENSMUST00000108391.8_path.png\n",
      "Generating report for ENSMUSG00000020836.15\n"
     ]
    }
   ],
   "source": [
    "sg.gen_report('Coro6',\n",
    "              'figures/coro6',\n",
    "              metadata_cols=['time_point'],\n",
    "              cmap='magma',\n",
    "              transcript_name=True,\n",
    "              novelty=True, \n",
    "              layer='pi', \n",
    "              browser=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "arbitrary-flower",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gid</th>\n",
       "      <th>p_val</th>\n",
       "      <th>dpi</th>\n",
       "      <th>adj_p_val</th>\n",
       "      <th>gname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>ENSMUSG00000026414.13</td>\n",
       "      <td>5.312136e-237</td>\n",
       "      <td>98.948153</td>\n",
       "      <td>2.645927e-234</td>\n",
       "      <td>Tnnt2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>ENSMUSG00000020836.15</td>\n",
       "      <td>1.787333e-64</td>\n",
       "      <td>97.932594</td>\n",
       "      <td>1.718034e-62</td>\n",
       "      <td>Coro6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>ENSMUSG00000020439.17</td>\n",
       "      <td>5.141485e-75</td>\n",
       "      <td>97.710327</td>\n",
       "      <td>5.523568e-73</td>\n",
       "      <td>Smtn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>946</th>\n",
       "      <td>ENSMUSG00000039542.16</td>\n",
       "      <td>7.076545e-128</td>\n",
       "      <td>96.704330</td>\n",
       "      <td>1.436015e-125</td>\n",
       "      <td>Ncam1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>ENSMUSG00000036918.16</td>\n",
       "      <td>3.520068e-06</td>\n",
       "      <td>95.447243</td>\n",
       "      <td>3.115743e-05</td>\n",
       "      <td>Ttc7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       gid          p_val        dpi      adj_p_val  gname\n",
       "281  ENSMUSG00000026414.13  5.312136e-237  98.948153  2.645927e-234  Tnnt2\n",
       "161  ENSMUSG00000020836.15   1.787333e-64  97.932594   1.718034e-62  Coro6\n",
       "151  ENSMUSG00000020439.17   5.141485e-75  97.710327   5.523568e-73   Smtn\n",
       "946  ENSMUSG00000039542.16  7.076545e-128  96.704330  1.436015e-125  Ncam1\n",
       "496  ENSMUSG00000036918.16   3.520068e-06  95.447243   3.115743e-05   Ttc7"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "die_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "provincial-hampton",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Plotting transcripts for ENSMUSG00000020439.17\n",
      "Saving transcript path graph for ENSMUST00000020721.14 as figures/smtn_browser_ENSMUST00000020721.14_path.png\n",
      "Saving transcript path graph for ENSMUST00000020718.9 as figures/smtn_browser_ENSMUST00000020718.9_path.png\n",
      "Saving transcript path graph for TALONT000451717 as figures/smtn_browser_TALONT000451717_path.png\n",
      "Saving transcript path graph for TALONT000451783 as figures/smtn_browser_TALONT000451783_path.png\n",
      "Saving transcript path graph for TALONT000451804 as figures/smtn_browser_TALONT000451804_path.png\n",
      "Saving transcript path graph for ENSMUST00000170588.7 as figures/smtn_browser_ENSMUST00000170588.7_path.png\n",
      "Generating report for ENSMUSG00000020439.17\n"
     ]
    }
   ],
   "source": [
    "sg.gen_report('Smtn',\n",
    "              'figures/smtn',\n",
    "              metadata_cols=['time_point'],\n",
    "              cmap='magma',\n",
    "              transcript_name=True,\n",
    "              novelty=True, \n",
    "              layer='pi', \n",
    "              browser=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "arabic-aggregate",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
